src/av2/evaluation/detection/utils.py

```diff
@@ -0,0 +1,439 @@
+
+    # Evaluate precision at different recalls.
+    precision_interpolated: NDArrayFloat = np.interp(recall_interpolated, recall, precision, right=0)  # type: ignore
+
+    average_precision: float = np.mean(precision_interpolated)
+    return average_precision, precision_interpolated
+
+
+def distance(dts: NDArrayFloat, gts: NDArrayFloat, metric: DistanceType) -> NDArrayFloat:
+    """Distance functions between detections and ground truth.
+
+    Args:
+        dts: (N,D) Detections where D is the number of attributes.
+        gts: (N,D) Ground truth labels where D is the number of attributes.
+        metric: Distance function type.
+
+    Returns:
+        (N,) Distance between the detections and ground truth under the specified metric.
+
+    Raises:
+        NotImplementedError: If the distance type is not supported.
+    """
+    if metric == DistanceType.TRANSLATION:
+        translation_errors: NDArrayFloat = np.linalg.norm(dts - gts, axis=1)  # type: ignore
+        return translation_errors
+    elif metric == DistanceType.SCALE:
+        scale_errors: NDArrayFloat = 1 - iou_3d_axis_aligned(dts, gts)
+        return scale_errors
+    elif metric == DistanceType.ORIENTATION:
+        yaws_dts: NDArrayFloat = mat_to_xyz(quat_to_mat(dts))[..., 2]
+        yaws_gts: NDArrayFloat = mat_to_xyz(quat_to_mat(gts))[..., 2]
+        orientation_errors = wrap_angles(yaws_dts - yaws_gts)
+        return orientation_errors
+    else:
+        raise NotImplementedError("This distance metric is not implemented!")
+
+
+def compute_objects_in_roi_mask(
+    cuboids_parameters: NDArrayFloat, city_SE3_ego: SE3, avm: ArgoverseStaticMap
+) -> NDArrayBool:
+    """Compute the evaluated cuboids mask based off whether _any_ of their vertices fall into the ROI.
+
+    Args:
+        cuboids_parameters: (N,10) Array of cuboid parameters corresponding to `ORDERED_CUBOID_COL_NAMES`.
+        city_SE3_ego: Egovehicle pose in the city reference frame.
+        avm: Argoverse map object.
+
+    Returns:
+        The boolean mask indicating which cuboids will be evaluated.
+    """
+    cuboid_list_ego: CuboidList = CuboidList([Cuboid.from_numpy(params) for params in cuboids_parameters])
+    cuboid_list_city = cuboid_list_ego.transform(city_SE3_ego)
+    cuboid_list_vertices_m = cuboid_list_city.vertices_m
+
+    is_within_roi = avm.get_raster_layer_points_boolean(
+        cuboid_list_vertices_m.reshape(-1, 3)[..., :2], RasterLayerType.ROI
+    )
+    is_within_roi = is_within_roi.reshape(-1, 8)
+    is_within_roi = is_within_roi.any(axis=1)
+    return is_within_roi
+
+
+def compute_evaluated_dts_mask(
+    dts: NDArrayFloat,
+    cfg: DetectionCfg,
+) -> NDArrayBool:
+    """Compute the evaluated cuboids mask.
+
+    Valid cuboids meet _two_ conditions:
+        1. The cuboid's centroid (x,y,z) must lie within the maximum range in the detection configuration.
+        2. The cuboid must have at _least_ one point in its interior.
+
```

Comment:
Docstring might be a bit confusing as we are not using the number of points condition in this function. Or should we add `is_valid_num_points: NDArrayBool = gts[:, -1] > 0` here too?